File: examples/plugin-repeat-ts/handler.ts
// Edit this file to implement your plugin logic.

import * as flatbuffers from 'flatbuffers';
import { RepeatResponse } from './plugin/fb/repeat/repeat-response';
import { BuildRepeatCallResponse, DecodeRepeatRequest } from './plugin/pluginart_helpers';

export function handle(payload: Buffer): Buffer {
  const request = DecodeRepeatRequest(payload);
  const input = request.payload.input() ?? '';
  const output = input.repeat(request.payload.count());

  const builder = new flatbuffers.Builder(128);
  const outputOff = builder.createString(output);
  RepeatResponse.startRepeatResponse(builder);
  RepeatResponse.addOutput(builder, outputOff);
  const response = RepeatResponse.endRepeatResponse(builder);
  return BuildRepeatCallResponse(request.call, builder, response);
}

File: examples/plugin-repeat-py/handler.py
from flatbuffers import Builder

from fb.repeat import RepeatResponse
from pluginart_helpers import BuildRepeatCallResponse, DecodeRepeatRequest

def handle(payload: bytes) -> bytes:
    request, call = DecodeRepeatRequest(payload)
    input_text = (request.Input() or b"").decode("utf-8")
    output = input_text * request.Count()

    builder = Builder(128)
    output_off = builder.CreateString(output)
    RepeatResponse.RepeatResponseStart(builder)
    RepeatResponse.RepeatResponseAddOutput(builder, output_off)
    response = RepeatResponse.RepeatResponseEnd(builder)
    return BuildRepeatCallResponse(call, builder, response)
